models/progresses.js

```diff
@@ -53,6 +55,22 @@ const getProgressDocument = async (queryParams) => {
   return progressDocs;
 };
 
+/**
+ * This function retrieves a paginated list of progress documents for a specific user or task,
+ * or for all users or all tasks if no specific user or task is provided.
+ * @param queryParams {object} This is the data that will be used for querying.
+ * It should be an object that includes key-value pairs for the fields - type, userId, taskId,
+ * and optionally pagination details such as page number and page size.
+ * @returns {Promise<object>} A Promise that resolves with the paginated progress document objects.
+ * @throws {Error} If the userId or taskId is invalid or does not exist.
+ **/
+const getPaginatedProgressDocument = async (queryParams) => {
+  await assertUserOrTaskExists(queryParams);
+  const page = queryParams.page || 0;
+  const query = buildQueryToFetchPaginatedDocs(queryParams);
+  const progressDocs = await getPaginatedProgressDocs(query, page);
+  return progressDocs;
```

Comment:
Shouldn't we wrap this inside a try-catch block?

Reply:
Since the controller where this function is being used already has a try-catch block to handle errors, I didn’t add an additional one inside the getPaginatedProgressDocument function.